network/dime_message.ksy

```diff
@@ -0,0 +1,61 @@
+meta:
+  id: dime_message
+  title: DIME (Direct Internet Message Encapsulation) Message
+  license: CC0-1.0
+  endian: be
+doc: |
+  Direct Internet Message Encapsulation (DIME)
+  is an old Microsoft specification for sending and receiving
+  SOAP messages along with additional attachments,
+  like binary files, XML fragments, and even other
+  SOAP messages, using standard transport protocols like HTTP.
+doc-ref: 
+  - http://xml.coverpages.org/draft-nielsen-dime-02.txt
+  - https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/december/sending-files-attachments-and-soap-messages-via-dime
+seq:
+  - id: dime
+    type: message
+types:
+  message:
+    seq:
+    - id: message
+      type: record
+      repeat: eos
+  record:
+    seq:
+    - id: version
+      type: b5
+    - id: first_record
+      type: b1
+    - id: last_record
+      type: b1
+    - id: chunck_record
+      type: b1
+    - id: type_format
+      type: b4
+    - id: reserved
+      type: b4
+    - id: options_length
+      type: u2
+    - id: id_length
+      type: u2
+    - id: type_length
+      type: u2
+    - id: data_length
+      type: u4
+    - id: options
+      size: options_length
+    - id: options_padding
+      size: (4 - _io.pos) % 4
```

Comment:
Padding can be a subtype. Power of 2 padding is better to implement without `%`.  Something like `(_io.pos + 3) & ~3 - _io.pos`

Reply:
> Power of 2 padding is better to implement without `%` (division is a costly operation).

IMHO these micro-optimizations are not worth it. To keep the code understandable and readable, we should primarily focus on expressing the intent.

For example, when the intent is to multiply `foo` by 63, you shouldn't do `(foo << 6) - foo`, but simply `foo * 63`.

If performance is a concern, choose a performant language like C++ and let the compiler do its job (it does a lot of transformations like from `foo * 63` to `(foo << 6) - foo` by itself). Additionally, if there is a great demand to have these for all languages, we can do them in the KSC as well, but I don't think it would really matter for most formats and usages.

Reply:
@generalmimon I missed your comment and I already made the changes, but I completely agree, readability is more important (especially with this kind of 2002 thing).

I would be for `(4 - _io.pos) % 4`. I will see what we will come out of the discussion and then either keep the newly commited `((_io.pos + 3) & ~3) - _io.pos` or go back to the original.